cloud/aws-common/src/main/java/org/apache/druid/common/aws/AWSClientUtil.java

```diff
@@ -22,11 +22,46 @@
 import com.amazonaws.AmazonClientException;
 import com.amazonaws.AmazonServiceException;
 import com.amazonaws.retry.RetryUtils;
+import com.amazonaws.services.s3.model.DeleteObjectsRequest;
+import com.amazonaws.services.s3.model.MultiObjectDeleteException;
+import com.google.common.collect.ImmutableSet;
 
 import java.io.IOException;
+import java.util.Set;
 
 public class AWSClientUtil
 {
+  /**
+   * This list of error code come from {@link RetryUtils}. At the moment, aws sdk does not expose a good
+   * way of retrying {@link com.amazonaws.services.s3.AmazonS3#deleteObjects(DeleteObjectsRequest)} requests. This
+   * request is used in org.apache.druid.storage.s3.S3DataSegmentKiller to delete a batch of segments from deep
+   * storage.
+   */
+  private static final Set<String> RECOVERABLE_ERROR_CODES = ImmutableSet.of(
```

Comment:
Can you add "InternalError" to this list?

Reply:
"ServiceUnavailable" too

Reply:
"SlowDown" and "503 SlowDown" too

Reply:
Basically all the 500, 502, 503, 504 from https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html

Reply:
thanks added